### DISCUSSION TITLE
Flexible Temporal Resolution

### DISCUSSION THREAD

**Discussion by clizbe** [TulipaEnergy/TulipaEnergyModel.jl#438] [Ideas]
_(From Germán)_
Here the list of topics to discuss:

•	Power constraints, which do not include durations (or maybe yes? E.g., weighted averages for VRE upper bound), written on the highest input/output resolution (This power constraints are not related at all to power-based piece-wise scheduling). 
•	Energy constraints, which include durations and are written on the lowest input/output resolution.
•	When do we have to impose energy or power constraints, otherwise the eqs. are simply wrong? I will show counter examples (hope I can prepare some)
•	Are there cases where the user can define either energy or power? I can only see cases where one is correct and the other is wrong, maybe I am missing something…
•	When do we have to impose some resolutions based on others? E.g., case of interconnection between two buses of the same commodity with different resolution… are there more cases?
•	What assets impose different resolution to that of the input/outputs (I suppose those ones involving other variables, e.g., SOC storage, UC, another?)
•	When should we check for the resolution of only input or only output but not both? [@Tejada Arango, D.A. (Diego)](mailto:diego.tejadaarango@tno.nl) do you remember? We talked about one case but I do not remember which one…
•	How to make a clear documentation? so people can easily follow. How to write the eqs on paper and in the documentation, we need to be very clear. I think the Matrix-way how we had it before is not the clearest way (it was more thought on how I would do it in AIMMS/GAMMS)
•	[@Greg Neustroev](mailto:G.Neustroev@tudelft.nl) would the sampling of the representative periods change if we know a priory the resolution to be used? Somehow I think so, should we take this into account?
•	[@Brafine, L.C. (Laura) <l.c.brafine@students.uu.nl>](mailto:l.c.brafine@students.uu.nl) & [@Greg Neustroev](mailto:G.Neustroev@tudelft.nl) before going to power-based with lower resolution (which is complicated when mixing resolutions), can we modify the energy profiles of the lower resolution so we do not loose (too much of) the ramp and peak information? Somehow I think so

**Comment by datejada** [comment]
After the meeting, we have updated what we need to do in the following issue https://github.com/TulipaEnergy/TulipaEnergyModel.jl/issues/344#issuecomment-1927330193
